site/testsuites/testsuite-kotlin/src/test/kotlin/docs/tbdex/pfi/pfi-overview.snippet.kt

```diff
@@ -0,0 +1,246 @@
+    }
+
+    override fun getExchanges(filter: GetExchangesFilter?): List<List<Message>> {
+        // Mock implementation for getExchanges
+        return listOf(
+            listOf(Message("Exchange 1"), Message("Exchange 2")),
+            listOf(Message("Exchange 3"))
+        )
+    }
+
+    override fun getRfq(exchangeId: String): Rfq {
+        return Rfq("RFQ for exchange $exchangeId")
+    }
+
+    override fun getQuote(exchangeId: String): Quote {
+        return Quote("Quote for exchange $exchangeId")
+    }
+
+    override fun getOrder(exchangeId: String): Order {
+        return Order("Order for exchange $exchangeId")
+    }
+
+    override fun getOrderStatuses(exchangeId: String): List<OrderStatus> {
+        // Mock implementation for getOrderStatuses
+        return listOf(OrderStatus("Status 1"), OrderStatus("Status 2"))
+    }
+
+    override fun getClose(exchangeId: String): Close {
+        return Close("Close information for exchange $exchangeId")
+    }
+
+    // :snippet-start: pfiOverviewWriteKt
+    suspend fun write(message: Message) {
+        val data = mapOf(
+            "exchangeid" to message.exchangeId,
+            "messagekind" to message.kind,
+            "messageid" to message.id,
+            "subject" to message.subject,
+            "message" to objectMapper.writeValueAsString(message)
+        )
+        dataProvider.insert("exchange", data)
+    }
+    // :snippet-end:
+}
+
+// :snippet-start: pfiOverviewReadOfferingsKt
+import tbdex.sdk.protocol.models.Offering
+
+class OfferingsApiProvider : OfferingsApi {
+    override fun getOffering(id: String): Offering {
+        val result = dataProvider.get("offering", opts.id).firstOrNull()
+        return if (result != null) Offering.factory(result["offering"] as JsonObject) else null
+    }
+
+    override fun getOfferings(filter: GetOfferingsFilter?): List<Offering> {
+        val results = dataProvider.query("offering", "*")
+        val offerings = mutableListOf<Offering>()
+
+        for (result in results) {
+            val offering = Offering.factory(result["offering"] as JsonObject)
+            offerings.add(offering)
+        }
+
+        return offerings
+    }
+// :snippet-end:
+
+    // :snippet-start: pfiOverviewWriteOfferingsKt
+    suspend fun create(offering: Offering) {
+        dataProvider.insert(
+            "offering",
+            mapOf(
+                "offeringid" to offering.id,
+                "payoutcurrency" to offering.payoutCurrency.currencyCode,
+                "payincurrency" to offering.payinCurrency.currencyCode,
+                "offering" to objectMapper.writeValueAsString(offering)
+            )
+        )
+    }
+    // :snippet-end:
+
+}
+
+// :snippet-start: pfiOverviewConfigKt
+import tbdex.sdk.httpserver.TbdexHttpServer
+import web5.sdk.crypto.AwsKeyManager
+import web5.sdk.dids.methods.dht.DidDht
+
+val serverConfig = TbdexHttpServerConfig(
+    port = 8080,
+    pfiDid = DidDht.create(AwsKeyManager()).uri,
```

Comment:
by time they are here, they already have a DID. let's just change this to `did.uri`

```suggestion
    pfiDid = did.uri,
```